File: frontend/src/i18n/languages.ts
/**
 * Configuration for supported languages in the frontend
 */

export interface Language {
  code: string;
  name: string;
  flag: string;
  isDefault?: boolean;
}

export const SUPPORTED_LANGUAGES: Language[] = [
  {
    code: 'pt-BR',
    name: 'Português (Brasil)',
    flag: '🇧🇷'
  },
  {
    code: 'en-US',
    name: 'English (United States)',
    flag: '🇺🇸',
    isDefault: true
  },
  {
    code: 'es-ES',
    name: 'Español (España)',
    flag: '🇪🇸'
  },
  {
    code: 'sv-SE',
    name: 'Svenska (Sverige)',
    flag: '🇸🇪'
  }
];

export const DEFAULT_LANGUAGE = 'en-US';

export const getLanguageByCode = (code: string): Language | undefined => {
  return SUPPORTED_LANGUAGES.find(lang => lang.code === code);
};

export const isValidLanguage = (code: string): boolean => {
  return SUPPORTED_LANGUAGES.some(lang => lang.code === code);
};

// Map i18next language codes to our format
export const LANGUAGE_CODE_MAP: Record<string, string> = {
  'pt-BR': 'pt-BR',
  'pt': 'pt-BR',
  'en-US': 'en-US',
  'en': 'en-US',
  'es-ES': 'es-ES',
  'es': 'es-ES',
  'sv-SE': 'sv-SE',
  'sv': 'sv-SE'
};

export const normalizeLanguageCode = (code: string): string => {
  return LANGUAGE_CODE_MAP[code] || DEFAULT_LANGUAGE;
};

File: (DEPRECATED)backend_django/locales/languages.py
"""
Configurações de idiomas suportados pelo Knight Agent
"""
from typing import List, Dict, Tuple

# Lista de idiomas suportados
SUPPORTED_LANGUAGES = [
    ('pt-BR', 'Português (Brasil)', '🇧🇷'),
    ('en-US', 'English (United States)', '🇺🇸'),
    ('es-ES', 'Español (España)', '🇪🇸'),
    ('sv-SE', 'Svenska (Sverige)', '🇸🇪'),
]

# Idioma padrão
DEFAULT_LANGUAGE = 'en-US'

# Mapeamento para códigos Django
DJANGO_LANGUAGE_CODES = {
    'pt-BR': 'pt-br',
    'en-US': 'en',
    'es-ES': 'es',
    'sv-SE': 'sv',
}

# Mapeamento reverso
FRONTEND_LANGUAGE_CODES = {
    'pt-br': 'pt-BR',
    'en': 'en-US',
    'es': 'es-ES',
    'sv': 'sv-SE',
}

def get_supported_languages() -> List[Tuple[str, str, str]]:
    """Retorna lista de idiomas suportados"""
    return SUPPORTED_LANGUAGES

def get_language_choices() -> List[Tuple[str, str]]:
    """Retorna choices para uso em models Django"""
    return [(code, f"{flag} {name}") for code, name, flag in SUPPORTED_LANGUAGES]

def get_django_code(frontend_code: str) -> str:
    """Converte código do frontend para Django"""
    return DJANGO_LANGUAGE_CODES.get(frontend_code, 'en')

def get_frontend_code(django_code: str) -> str:
    """Converte código do Django para frontend"""
    return FRONTEND_LANGUAGE_CODES.get(django_code, 'en-US')

def is_supported_language(language_code: str) -> bool:
    """Verifica se um idioma é suportado"""
    return language_code in [code for code, _, _ in SUPPORTED_LANGUAGES]